tools/llvm-kompile-compute-loc/main.cpp

```diff
@@ -60,17 +48,44 @@ get_kore_location(std::string &definition, int const &ordinal) {
   return std::nullopt;
 }
 
+std::optional<std::pair<std::string, int64_t>>
+get_k_location(std::string &definition, int const &ordinal) {
+  // Parse the definition.kore to get the AST.
+  kllvm::parser::kore_parser parser(definition);
+  auto kore_ast = parser.definition();
+  kore_ast->preprocess();
+
+  auto axiom = kore_ast->get_axiom_by_ordinal(ordinal);
+  auto result = get_start_line_location(axiom);
+  if (result) {
+    return result;
+  }
+  auto result2 = get_kore_location(definition, ordinal);
+  if (result2) {
+    return std::make_pair<std::string, uint64_t>("", *result2);
+  }
+  return std::nullopt;
+}
+
 int main(int argc, char **argv) {
   cl::HideUnrelatedOptions({&loc_cat});
   cl::ParseCommandLineOptions(argc, argv);
 
   auto definition = kompiled_dir + "/definition.kore";
 
-  auto location = is_k_line ? get_k_location(definition, ordinal)
-                            : get_kore_location(definition, ordinal);
+  std::optional<std::pair<std::string, uint64_t>> location;
+  if (is_kore_line) {
+    auto line = get_kore_location(definition, ordinal);
+    location = std::make_pair(definition, *line);
+  } else {
+    location = get_k_location(definition, ordinal);
+    if (location->first.empty()) {
+      location->first = definition;
+    }
```

Comment:
Why we're testing this and doing this assignment instead of directly returning the definition in `get_k_location`?